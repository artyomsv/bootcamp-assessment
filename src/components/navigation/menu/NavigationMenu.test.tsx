import * as React from 'react';
import {createShallow} from '@material-ui/core/test-utils';
import {createMount} from '@material-ui/core/test-utils';
import NavigationMenu from './NavigationMenu.widget';
import MenuItem from '@material-ui/core/MenuItem';

const shallow = createShallow();
const mount = createMount();

describe('renders without crashing', () => {
  test('if renders', () => {
    const mockFn = jest.fn();
    const result = shallow(<NavigationMenu navigateTo={mockFn} selectedPage={'home'}/>);
    expect(result).toMatchSnapshot();
  });
});

describe('Test Navigation', () => {
  test('if renders', () => {
    const mockFn = jest.fn();
    const wrap = mount(<NavigationMenu navigateTo={mockFn} selectedPage={'home'}/>);
    expect(wrap.find(MenuItem).length).toBe(3);
  });
});
