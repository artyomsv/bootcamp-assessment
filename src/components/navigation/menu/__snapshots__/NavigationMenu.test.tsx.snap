// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing if renders 1`] = `
<Navigation
  classes={
    Object {
      "root": "Navigation-root-1",
    }
  }
  navigateTo={[MockFunction]}
/>
`;
